bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/link/contentassist/CompletionProposalPopup2.java

```diff
@@ -872,18 +872,14 @@ public void processEvent(VerifyEvent e) {
 	private void filterProposals() {
 		++ fInvocationCounter;
 		Control control= fViewer.getTextWidget();
-		control.getDisplay().asyncExec(new Runnable() {
+		control.getDisplay().asyncExec(() -> {
 			long fCounter= fInvocationCounter;
```

Comment:
Before, fCounter is initialized just once, when `filterProposals` is invoked and the `Runnable` is created, ie before actually running the Runnable. While with the change, the value is set at execution of the Runnable, leading to...

Reply:
will moving `fCounter`  assignment outside of the lambda solve the problem?

Reply:
Yes, I think it would.

Reply:
@mickaelistria I have made that change accordingly and removed the problematic lambda. please have a look.